### PR TITLE
Extending partial methods

### PR DESCRIPTION
**Pull request by jaredpar**

### CODE REVIEW

**Comment by RUSshy** [comment]
Will there be some methods available in the standard library to help with compile time reflection? or it is a non goal?

File: proposals/extending-partial-methods.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+}
+
+partial class C
+{
+    // The definition of C.M
+    partial void M(string message) => Console.WriteLine(message);
+}
+```
+
+One behavior of `partial` methods is that when the definition is abscent then
```
Comment (jnm2) [inline proposals/extending-partial-methods.md:28]:
```suggestion
One behavior of `partial` methods is that when the definition is absent then
```

File: proposals/extending-partial-methods.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    {
+        M(GetIt()); // Call to M and GetIt erased at compile time
+    }
+
+    string GetIt() => "Hello World";
+}
+```
+
+The original motivation for this feature was source generation in the form of 
+designer generated code. Users were constantly editting the generated code
```
Comment (jnm2) [inline proposals/extending-partial-methods.md:48]:
```suggestion
designer generated code. Users were constantly editing the generated code 
```

**Comment by jaredpar** [comment]
> Will there be some methods available in the standard library to help with compile time reflection? or it is a non goal?

Can you elaborate a bit on what you're asking here? Not quite sure what functionality you're referring to.

**Pull request merged**